### PR TITLE
Jdc provide missing transactions

### PR DESCRIPTION
**Pull request by lorbax**
This PR is reviewable, I will remove WIP as soon as I write a MG test for checking that the code works
Reviewed, MG test postponed to another PR, this is addressed in issue [661](https://github.com/stratum-mining/stratum/issues/661)

### CODE REVIEW

File: roles/jd-client/src/job_declarator/mod.rs [hunk 1]
```diff
@@ -201,15 +203,22 @@ impl JobDeclarator {
...
         let (id, _, sender) = self_mutex
             .safe_lock(|s| (s.req_ids.next(), s.min_extranonce_size, s.sender.clone()))
             .unwrap();
         // TODO: create right nonce
         let tx_short_hash_nonce = 0;
+        let mut tx_list: Vec<Transaction> = Vec::new();
+        for tx in tx_list_sequence.to_vec() {
+            //TODO remove unwrap
+            //TODO check if we want deserialize or consensus_decode
+            let tx = Transaction::deserialize(&tx).unwrap();
```
Comment (lorbax) [inline roles/jd-client/src/job_declarator/mod.rs:219]: @Fi3 it is fine here to use deserialize? should I use consensus_decode?

File: protocols/v2/roles-logic-sv2/src/utils.rs [hunk 2]
```diff
@@ -665,14 +665,13 @@ pub fn get_target(
...
 }
 pub fn hash_lists_tuple(
-    tx_data: Seq064K<'static, B016M<'static>>,
+    tx_data: Vec<Transaction>,
     tx_short_hash_nonce: u64,
 ) -> (Seq064K<'static, ShortTxId<'static>>, U256<'static>) {
     let mut txid_list: Vec<bitcoin::Txid> = Vec::new();
-    for tx in tx_data.to_vec() {
+    for tx in tx_data {
         //TODO remove unwrap
```
Comment (Fi3) [inline protocols/v2/roles-logic-sv2/src/utils.rs:673]: you can remove this TODO

File: roles/jd-client/src/job_declarator/mod.rs [hunk 3]
```diff
@@ -201,15 +203,22 @@ impl JobDeclarator {
         self_mutex: &Arc<Mutex<Self>>,
         template: NewTemplate<'static>,
         token: Vec<u8>,
-        tx_list: Seq064K<'static, B016M<'static>>,
+        tx_list_sequence: Seq064K<'static, B016M<'static>>,
```
Comment (Fi3) [inline roles/jd-client/src/job_declarator/mod.rs:206]: no need to have list and sequence in the name chose one and stick with it: `tx_list` or `tx_sequence`.

Personally I prefer `tx_list`.

File: roles/jd-client/src/job_declarator/mod.rs (hunk 1 above)
Comment (Fi3) [reply roles/jd-client/src/job_declarator/mod.rs:219]: deserialize call consensus_decode_from_finite_reader so is ok

Comment (Fi3) [reply roles/jd-client/src/job_declarator/mod.rs:219]: add an issue to add logic that manage unhappy path instead of unwrapping.

**Pull request merged**